javatests/arcs/android/host/AndroidEntityHandleManagerTest.kt

```diff
@@ -227,22 +222,22 @@ class AndroidEntityHandleManagerTest : LifecycleOwner {
         handleName: String,
         handleMode: HandleMode
     ) = handleManager.createSingletonHandle(
-        handleHolder.getEntitySpec(handleName),
+        handleMode,
         handleName,
+        handleHolder.getEntitySpec(handleName),
         singletonKey,
-        schema,
-        handleMode
+        schema
     ).also { handleHolder.setHandle(handleName, it) }
 
-    private suspend fun createSetHandle(
+    private suspend fun createCollectionHandle(
         handleManager: EntityHandleManager,
         handleName: String,
         handleMode: HandleMode
-    ) = handleManager.createSetHandle(
-        handleHolder.getEntitySpec(handleName),
+    ) = handleManager.createCollectionHandle(
+        handleMode,
         handleName,
+        handleHolder.getEntitySpec(handleName),
         setKey,
-        schema,
-        handleMode
+        schema
     ).also { handleHolder.setHandle(handleName, it) }
 }
```

Comment:
Can you add a test for checkCanRead/checkCanWrite()?

Reply:
Done